src/pybind/mgr/ansible/module.py

```diff
@@ -288,18 +304,61 @@ def get_inventory(self, node_filter=None):
 
         return ansible_operation
 
-    def create_osds(self, osd_spec):
+    def create_osds(self, osd_spec, host=None):
```

Comment:
Can't we have a function that does an `ansible_operation`?  `create_osds` and `remove_osds` are 95% identical the main difference is the playbook being used and some other things for `remove_osds`. This will avoid duplicating the same code all the time we add a new operation.

Reply:
I make the design with this target  :-), all the methods implementing operations with the same structure:
1- Process of parameters ( different in each method)
2- Instance a new completion object ( slightly different in each method)
3- How to process the result (different in each method) 
4- Launch/store the operation using the completion object (The same in all methods).

I think that we can ·extract" the task in point 4, and make them common to all the methods that implemeny orchestrator operations.

But the rest of tasks i think that is good to leave them where they are( readibility, easiness), to move them is just to move the same code to another place, introducing a new "layer" that only make that in the operational methods  will have less code.

Thank you for your idea!, i will use it to simplify the common part ( task 4)

Reply:
Sounds good, thanks

Reply:
Improvement included. Thx!